ishop3/src/components/Product.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component } from 'react';
+
+export default class Product extends Component {
+  deleteProduct = event => {
+    let deleteQuestion = confirm(`Вы действительно хотите удалить товар ${this.props.name}?`);
```

Comment:
`confirm` непредсказуемо работает, видимо, он мешает процессу рендера, поэтому обернём его в `setTimeout` с 0 задержкой, чтобы `confirm` 100% был вызван после `this.highlightProduct()`. Если знать `event loop` и то, как работает код в js, что идёт первым, что вторым, как работает асинхронный код, то можно понять, почему `setTimeout` вызывается после обычного синхронного `this.highlightProduct()`.

![image](https://user-images.githubusercontent.com/55033464/105509351-75931280-5cde-11eb-8475-45a83c244425.png)



  ```
deleteProduct = event => {
    event.stopPropagation();
    let deleteQuestion = 0;

    setTimeout(()=>{
      deleteQuestion = confirm(`Вы действительно хотите удалить товар ${this.props.name}?`);
    }, 0)

    this.highlightProduct();
    if (deleteQuestion) {
      this.props.filterArray(this.props.code);
      this.props.changeMode('hidden');
    }
  };
```

Reply:
А ведь приведённый код не работает. Сначала включится проверка на `deleteQuestion`, и только после этого откроется confirm.
Если завернуть условие в `setTimeout`, то всё начинает работать как надо:

```suggestion
      setTimeout(() => {
        deleteQuestion = confirm(`Вы действительно хотите удалить товар ${this.props.name}?`);

        if (deleteQuestion) {
          this.props.filterArray(this.props.code);
          this.props.changeMode('hidden');
       }
    }, 0);
```